skorch/net.py

```diff
@@ -1421,6 +1421,22 @@ def save_params(
         if f_history is not None:
             self.history.to_file(f_history)
 
+    def _get_state_dict(self, f):
+        """Load torch object (module, optimizer) using desired device."""
+        map_location = torch.device(self.device)
+
+        # The user wants to use CUDA but there is no CUDA device
+        # available, thus fall back to CPU.
+        if self.device.startswith('cuda') and not torch.cuda.is_available():
+            warnings.warn(
+                "Model configured to use CUDA but no CUDA devices "
+                "available. Loading on CPU instead.",
+                ResourceWarning)
+            self.device = 'cpu'
+            map_location = torch.device('cpu')
+
+        return torch.load(f, map_location=map_location)
+
```

Comment:
Two things:
1. The warning should include that the `self.device` parameter is now set to `'cpu'`.
2. I think we should refactor `_get_state_dict` into `_get_map_location` so it can be used in `__setstate__` as well since the code there is doing basically the same thing.